Outlook spam reporting sample (#675)
OfficeDev/Office-Add-in-samples: Samples/outlook-spam-reporting/README.md

---
title: "Report spam or phishing emails in Outlook (preview)"
page_type: sample
urlFragment: outlook-spam-reporting
products:
- office-add-ins
- office-outlook
- office
languages:
- javascript
extensions:
  contentType: samples
  technologies:
  - Add-ins
  createdDate: 02/12/2024 10:00:00 AM
description: "Learn how to create an integrated spam-reporting add-in in Outlook."
---

# Report spam or phishing emails in Outlook (preview)

**Applies to**: Outlook on Windows (preview)

> [A sample spam-reporting dialog.](/assets/readme/outlook-spam-processing-dialog.png)

Reviewer: ```suggestion
![A sample spam-reporting dialog.](/assets/readme/outlook-spam-processing-dialog.png)
```

> ![A sample spam-reporting dialog.](/assets/readme/outlook-spam-processing-dialog.png)

Reviewer: ```suggestion
![A sample spam-reporting dialog.](./assets/readme/outlook-spam-processing-dialog.png)
```

## Summary

This sample showcases how to build an integrated spam-reporting solution that:

- Is easily discoverable in the Outlook client ribbon.
- Provides the user with a processing dialog to report an email.
- Facilitates saving a copy of the reported email to a file to submit it to your backend system for further processing.

To learn about key components of this sample, see [Implement an integrated spam-reporting add-in (preview)](https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting).

> > [!IMPORTANT]
> 
> > The integrated spam-reporting feature is currently in preview in Outlook on Windows. Features in preview shouldn't be used in production add-ins. We invite you to try out this feature in test or development environments and welcome feedback on your experience through [GitHub](https://github.com/OfficeDev/office-js/issues/new/choose).

Reviewer: ```suggestion
> [!IMPORTANT]
> The integrated spam-reporting feature is currently in preview in Outlook on Windows. Features in preview shouldn't be used in production add-ins. We invite you to try out this feature in test or development environments and welcome feedback on your experience through [GitHub](https://github.com/OfficeDev/office-js/issues/new/choose).
```

## Applies to

Outlook on Windows starting in Version 2307 (Build 16626.10000).

>     > [!NOTE]
>     > If you don't have a Microsoft 365 subscription, you might qualify for a free developer subscription that's renewable for 90 days and comes configured with sample data. For details, see the [Microsoft 365 Developer Program FAQ](https://learn.microsoft.com/office/developer-program/microsoft-365-developer-program-faq#who-qualifies-for-a-microsoft-365-e5-developer-subscription-).

Reviewer: ```suggestion

> [!NOTE]
> If you don't have a Microsoft 365 subscription, you might qualify for a free developer subscription that's renewable for 90 days and comes configured with sample data. For details, see the [Microsoft 365 Developer Program FAQ](https://learn.microsoft.com/office/developer-program/microsoft-365-developer-program-faq#who-qualifies-for-a-microsoft-365-e5-developer-subscription-).
```

## Prerequisites
- Microsoft 365 subscription

>     > [!TIP]
>     > If you're unable to choose a channel in your Outlook client, see [Let users choose which Microsoft 365 Insider channel to install on Windows devices](https://learn.microsoft.com/deployoffice/insider/deploy/user-choice).

Reviewer: ```suggestion
> [!TIP]
> If you're unable to choose a channel in your Outlook client, see [Let users choose which Microsoft 365 Insider channel to install on Windows devices](https://learn.microsoft.com/deployoffice/insider/deploy/user-choice).
```

> [!IMPORTANT]
> To test the `getAsFileAsync` method while it's still in preview in Outlook on Windows, you must configure your computer's registry.
>
> Outlook on Windows includes a local copy of the production and beta versions of Office.js instead of loading from the content delivery network (CDN). By default, the local production copy of the API is referenced. To reference the local beta copy of the API, you must configure your computer's registry as follows:
>
> 1. In the registry, navigate to `HKEY_CURRENT_USER\SOFTWARE\Microsoft\Office\16.0\Outlook\Options\WebExt\Developer`. If the key doesn't exist, create it.
> 1. Create an entry named `EnableBetaAPIsInJavaScript` and set its value to `1`.
>

> >    :::image type="content" source="../images/outlook-beta-registry-key.png" alt-text="The EnableBetaAPIsInJavaScript registry value is set to 1.":::

Reviewer: ```suggestion
>    ![The EnableBetaAPIsInJavaScript registry value is set to 1.](https://learn.microsoft.com/office/dev/add-ins/images/outlook-beta-registry-key.png)
```

   ```console
    npx office-addin-dev-certs install
    ```
	
    This command will display the folder location where it generated the certificate files.
	
1. Go to the folder location where the certificate files were generated, then copy the **localhost.crt** and **localhost.key** files to the cloned or downloaded sample folder.
1. Run the following command.

    ```console
    http-server -S -C localhost.crt -K localhost.key --cors . -p 3000
    ```
	
    The http-server will run and host the current folder's files on localhost:3000.
	
1. Now that your localhost web server is running, you can sideload the **manifest-localhost.xml** file provided in the sample folder. To sideload the manifest, follow the manual instructions in [Sideload Outlook add-ins for testing](https://learn.microsoft.com/office/dev/add-ins/outlook/sideload-outlook-add-ins-for-testing?tabs=windows-web#sideload-manually).
1. Follow the steps in [Try it out](#try-it-out) to test the sample.

## Try it out

Once the add-in is loaded in Outlook, use the following steps to try out its functionality.

1. Choose a message from your inbox, then select the add-in's button from the ribbon.

>     ![The spam-reporting add-in button is selected from the ribbon.](/assets//readme/outlook-spam-ribbon-button.png)

Reviewer: ```suggestion
    ![The spam-reporting add-in button is selected from the ribbon.](./assets//readme/outlook-spam-ribbon-button.png)
```

>     [The spam-reporting add-in button is selected from the ribbon.](/assets//readme/outlook-spam-ribbon-button.png)

Reviewer: ```suggestion
    ![The spam-reporting add-in button is selected from the ribbon.](/assets//readme/outlook-spam-ribbon-button.png)
```

1. In the preprocessing dialog, choose a reason for reporting the message and add information about the message, if configured. Then, select **Report**.
1. (Optional) In the post-processing dialog, select **OK**.

>     ![The post-processing dialog of the sample spam-reporting add-in.](/assets//readme/outlook-spam-post-processing-dialog.png)

Reviewer: ```suggestion
    ![The post-processing dialog of the sample spam-reporting add-in.](./assets//readme/outlook-spam-post-processing-dialog.png)
```

>     [The post-processing dialog of the sample spam-reporting add-in.](/assets//readme/outlook-spam-post-processing-dialog.png)

Reviewer: ```suggestion
    ![The post-processing dialog of the sample spam-reporting add-in.](/assets//readme/outlook-spam-post-processing-dialog.png)
```
